### PR TITLE
Error log when license verification fails locally

### PR DESCRIPTION
**Pull request by albertzaharovits**
This change implements logging if the license verification fails on local nodes
(after the the license has already been (erroneously) published in the cluster state).

This also contains some arguable minor code refactoring that I hope help with code readability.

### CODE REVIEW

**Comment by albertzaharovits** [comment]
Note to reviewers: I recommend reviewing the PRs in order.
The second one https://github.com/elastic/elasticsearch/pull/102919/commits/cb3ad2efa1c1d958b1041eb043632a9afa84fae7 is probably the only one that **might** actually fix something.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/license/ClusterStateLicenseService.java [hunk 1]
```diff
@@ -479,6 +452,36 @@ private void updateXPackLicenseState(License license) {
         }
     }
 
+    private void maybeRegisterOrUpdateLicense(ClusterState previousClusterState, ClusterState currentClusterState) {
+        if (XPackPlugin.isReadyForXPackCustomMetadata(currentClusterState) == false) {
```
Comment (jakelandis) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/license/ClusterStateLicenseService.java:456]: +1 to completely remove this whole if block if you choose to ... i started to remove these across the board... but kinda forgot about it: https://github.com/elastic/elasticsearch/pull/95428

**Review by jakelandis — APPROVED** [review]
LGTM

File: x-pack/plugin/core/src/main/java/org/elasticsearch/license/ClusterStateLicenseService.java (hunk 1 above)
Comment (albertzaharovits) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/license/ClusterStateLicenseService.java]: Ah, thank you for the indication!
I also wanted to ask about it, but similarly omitted....
I've pushed [0c360c2](https://github.com/elastic/elasticsearch/pull/102919/commits/0c360c29d6f86369dce96e65c9684468be622d88) .

**Pull request merged**